### PR TITLE
Make `Authorizer` pluggable

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

File: servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/QuarkusAuthorizer.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  @Inject
+  public QuarkusAuthorizer(
+      QuarkusNessieAuthorizationConfig config, @Any Instance<Authorizer> authorizers) {
+    if (config.enabled()) {
+      if (authorizers.isUnsatisfied()) {
+        throw new IllegalStateException("No Nessie Authorizer available");
+      }
+      Instance<Authorizer> authorizerInstance =
+          authorizers.select(new AuthorizerType.Literal(config.authorizationMode()));
```
Comment (dimas-b) [inline servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/QuarkusAuthorizer.java:46]: "type" vs. "mode" looks a bit odd. WDYT about renaming config to `config.authorizerType()`?

**Review by dimas-b — DISMISSED** [review]

File: servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/QuarkusAuthorizer.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+  }
+
+  @Override
+  public BatchAccessChecker startAccessCheck(AccessContext context) {
+    return this.authorizer.startAccessCheck(context);
+  }
+
+  public void eagerAuthorizerInitialization(
+      @Observes StartupEvent event, @SuppressWarnings("unused") Authorizer authorizer) {
```
[the remark this answers is not in the document]
Comment (snazy) [reply servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/QuarkusAuthorizer.java:66]: That's exactly the intent here

[the remark this answers is not in the document]
Comment (snazy) [reply servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/QuarkusAuthorizer.java:66]: It _shall_ initialize it to run into possible config issues early

[the remark this answers is not in the document]
Comment (dimas-b) [reply servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/QuarkusAuthorizer.java:66]: That initialization will happen in the `QuarkusAuthorizer` constructor anyway. The `Authorizer` reference here will probably cause a CDC resolution error if there are more that one instance available.

[the remark this answers is not in the document]
Comment (adutra) [reply servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/QuarkusAuthorizer.java:66]: Shouldn't the class be annotated with `@Startup` instead?

https://quarkus.io/version/main/guides/lifecycle#startup_annotation

**Review by dimas-b — APPROVED** [review]

**Pull request merged**